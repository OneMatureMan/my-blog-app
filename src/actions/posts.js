import { v1 as uuidv1 } from 'uuid';
import moment from 'moment';

export const addPost = (
    {
        title = '',
        body = '',
        createdAt = 0
    } = {}
) => ({
    type: 'ADD_POST',
    post : {
        id: uuidv1(),
        title,
        body,
        createdAt: moment()
    }
})

export const editPost = (id, updates) => ({
    type: 'EDIT_POST',
    id,
    updates
});


export const removePost = ({ id } = {}) => ({
    type: 'REMOVE_POST',
    id
})