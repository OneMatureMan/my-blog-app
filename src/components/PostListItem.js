import React from 'react';
import { Link } from 'react-router-dom';

const PostListItem = ({ id, title, body, createdAt }) => (
    <Link to={`edit/${id}`}>
        <p>{title} - {createdAt.valueOf()}</p>
    </Link> 
);

export default PostListItem;