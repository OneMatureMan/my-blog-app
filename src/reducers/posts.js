const postsDefaultState = []

const postsReducer = (state = postsDefaultState, action) => {
    switch (action.type) {
        case 'ADD_POST':
            return state.concat({ id:action.id, ...action.post});
        case 'EDIT_POST':
            return state.map(post => {
                if (post.id === action.id) {
                    return {
                        ...post,
                        ...action.updates
                    }
                } else {
                    return post;
                }
            })
        case 'REMOVE_POST':
            return state.filter(({ id }) => id !== action.id)
        default :
            return state;
    }
}

export default postsReducer;