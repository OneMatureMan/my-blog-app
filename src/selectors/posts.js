// get visible posts
export default (posts, { text, sortBy }) => {
    return posts.filter(({ title }) => title.toLowerCase().includes(text.toLowerCase()))
            .sort((a,b) => {
                if (sortBy === 'date') {
                    if (a.createdAt < b.createdAt) {
                        return 1;
                    }
                    if (a.createdAt > b.createdAt) {
                        return -1;
                    }
                    return 0;
                } else if (sortBy === 'title') {
                    
                    if (a.title.toLowerCase() < b.title.toLowerCase()) {
                        return -1;
                    }
                    
                    if (a.title.toLowerCase() > b.title.toLowerCase()) {
                        return 1;
                    }
                    return 0;
                }            
            })
}