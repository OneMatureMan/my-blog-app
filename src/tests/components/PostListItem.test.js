import React from 'react';
import PostListItem from '../../components/PostListItem';
import posts from '../fixtures/posts';
import { shallow } from 'enzyme';


test('should render PostListItem component correctly', () => {
    const wrapper = shallow(<PostListItem {...posts[1]} />);
    expect(wrapper).toMatchSnapshot();
})