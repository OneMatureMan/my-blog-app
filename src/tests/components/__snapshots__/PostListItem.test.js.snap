// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PostListItem component correctly 1`] = `
<Link
  replace={false}
  to="edit/2"
>
  <p>
    Going to the Store on Mondays
     - 
    3000
  </p>
</Link>
`;
